keen/tests/access_key_tests.py

```diff
@@ -0,0 +1,111 @@
+
+    NO_CONTENT_RESPONSE = MockedResponse(status_code=204, json_response="")
+
+    def setUp(self):
+        super(AccessKeyTests, self).setUp()
+        keen.project_id = "55777979e085574e8ad3523c"
+        keen.write_key = "DEADBEEF"
+        keen.read_key = "BADFEED"
+        keen.master_key = "BADHORSE"
+        self.keys_uri_prefix = "https://api.keen.io/3.0/projects/{0}/keys".format(keen.project_id)
+
+    def _assert_proper_permissions(self, method, permission):
+        self.assertTrue(permission in method.call_args[1]["headers"]["Authorization"])
+
+    @patch("requests.Session.post")
+    def test_create_access_key(self, post):
+        post.return_value = self.ACCESS_KEY_RESPONSE
+        resp = keen.create_access_key(self.ACCESS_KEY_NAME)
+        self.assertTrue(self.ACCESS_KEY_NAME in post.call_args[1]["data"])
+        self._assert_proper_permissions(post, keen.master_key)
+        self.assertEqual(resp, self.ACCESS_KEY_RESPONSE.json())
+
+    @patch("requests.Session.get")
+    def test_list_access_keys(self, get):
+        get.return_value = self.ACCESS_KEY_RESPONSE
+        resp = keen.list_access_keys()
+        self.assertEqual(self.keys_uri_prefix, get.call_args[0][0])
+        self._assert_proper_permissions(get, keen.master_key)
+        self.assertEqual(resp, self.ACCESS_KEY_RESPONSE.json())
+
+    @patch("requests.Session.get")
+    def test_get_access_key(self, get):
+        get.return_value = self.ACCESS_KEY_RESPONSE
+        resp = keen.get_access_key(self.ACCESS_KEY_NAME)
+        self.assertEqual("{0}/{1}".format(self.keys_uri_prefix, self.ACCESS_KEY_NAME), get.call_args[0][0])
+        self._assert_proper_permissions(get, keen.master_key)
+        self.assertEqual(resp, self.ACCESS_KEY_RESPONSE.json())
+
+    @patch("requests.Session.post")
+    def test_revoke_access_key(self, post):
+        post.return_value = self.NO_CONTENT_RESPONSE
+        resp = keen.revoke_access_key(self.ACCESS_KEY_NAME)
+        self.assertEqual("{0}/{1}/revoke".format(self.keys_uri_prefix, self.ACCESS_KEY_NAME), post.call_args[0][0])
+        self._assert_proper_permissions(post, keen.master_key)
+        self.assertEqual(resp, self.NO_CONTENT_RESPONSE.json())
+
+    @patch("requests.Session.post")
+    def test_unrevoke_access_key(self, post):
+        post.return_value = self.NO_CONTENT_RESPONSE
+        resp = keen.unrevoke_access_key(self.ACCESS_KEY_NAME)
+        self.assertEqual("{0}/{1}/unrevoke".format(self.keys_uri_prefix, self.ACCESS_KEY_NAME), post.call_args[0][0])
+        self._assert_proper_permissions(post, keen.master_key)
+        self.assertEqual(resp, self.NO_CONTENT_RESPONSE.json())
+
+    @patch("requests.Session.post")
+    def test_update_access_key_full(self, post):
+        # The update tests have a significant amount of logic that will not be tested via blackbox testing without
+        # un-mocking Keen's API. So this is the only test that will really cover any of them, and not even very
+        # well.
```

Comment:
All of these tests seem to only make sure the URL is generated properly and that the client/api regurgitate the response that's given to the mock. This gets good CC numbers, but isn't really doing too much in terms of being more certain it'll work with the real back end. I realize that's how all the query tests and most of the existing tests in general are, and it's out of scope for this PR, but I wonder if we shouldn't have a push a to do some validation on the payload generated like we do in `test_update_full_saved_query()` and friends (see commit c4e3cae). If that seems like a good idea, I think it'd be worth filing an issue.